cmd/util/auth/ask.go

```diff
@@ -45,26 +45,46 @@ func askResponder(cmd *cobra.Command) responder {
 
 			fmt.Fprintf(cmd.ErrOrStderr(), "%s: ", name)
 
-			var input []byte
-			var err error
-
-			// If the property is marked as write only, assume it is a sensitive
-			// value and make sure we don't display it in the terminal
-			if subschema.WriteOnly && term.IsTerminal(int(os.Stdin.Fd())) {
-				input, err = term.ReadPassword(int(os.Stdin.Fd()))
-				fmt.Fprintln(cmd.ErrOrStderr())
-			} else {
-				reader := bufio.NewScanner(cmd.InOrStdin())
-				if reader.Scan() {
-					input = reader.Bytes()
+			inputChan := make(chan []byte, 1)
+			errChan := make(chan error, 1)
+
+			// Reading a value is not a cancellable operation. So we manually
+			// listen for the context here and write a final byte to the input
+			// if the user cancels, to hopefully get the input read to end.
+			go func(inputCh chan<- []byte, errCh chan<- error) {
+				defer close(inputCh)
+				defer close(errCh)
```

Comment:
Probably safest not to close these channels, we only need to do that to let the receiver know there are no more messages coming.  If we close them here, I'm worried about an ordering of instructions like

* Write to inputCh 
* Close inputCh
* Select on inputCh 

If we close them outside the goroutine, we might accidentally write to a closed channel.  
So safest to let the GC dispose of them.

Reply:
Ok, I went and checked my understanding about channels, and I think it would be fine to select on a closed channel as long as we used the `value, ok := <-chan` idiom to check for channel closure. 

But also, just allowing them to go out of scope should work fine here too because we will always push something to one of the channels, so I'll just do that.